samples/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -380,16 +463,24 @@ static bool event_handler(const struct event_header *eh)
 {
 	if (is_module_state_event(eh)) {
 		const struct module_state_event *event = cast_module_state_event(eh);
+		static int err;
+
+		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			/* Settings initialized before config module */
+			err = init_settings();
+		}
 
 		if (check_state(event, MODULE_ID(config), MODULE_STATE_READY)) {
 			static bool initialized;
 
 			__ASSERT_NO_MSG(!initialized);
 			initialized = true;
 
-			init();
-
-			module_set_state(MODULE_STATE_READY);
+			if (!init() && !err) {
+				module_set_state(MODULE_STATE_READY);
+			} else {
+				module_set_state(MODULE_STATE_ERROR);
```

Comment:
let's assume error can be set multiple times - we will handle it when needed

Reply:
ok, changed